pkg/cli/completion.go

```diff
@@ -0,0 +1,82 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cli
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/spf13/cobra"
+)
+
+func (c *cli) newCompletionCmd() *cobra.Command {
+	return &cobra.Command{
+		Use:                   "completion [bash|zsh|powershell]", // TODO (requires spf13.cobra update): add fish
```

Comment:
It usage/description is not so good as before. See: 

**master**

```
 $ kubebuilder completion 
Output shell completion code for the specified shell (bash or zsh). 
The shell code must be evaluated to provide interactive completion of kubebuilder commands.
This can be done by sourcing ~/.bash_profile or ~/.bashrc.
Detailed instructions on how to do this are available at docs/book/src/reference/completion.md

Usage:
  kubebuilder completion [command]

Examples:
To load all completions run:
$ . <(kubebuilder completion)
To configure your shell to load completions for each session add to your .bashrc:
$ echo -e "\n. <(kubebuilder completion)" >> ~/.bashrc


Available Commands:
  bash        Generate bash completions
  zsh         Generate zsh completions

Flags:
  -h, --help   help for completion

Use "kubebuilder completion [command] --help" for more information about a command.
```

Now: 

```
 $ kubebuilder completion 
Error: accepts 1 arg(s), received 0
Usage:
  kubebuilder completion [bash|zsh|powershell]

Flags:
  -h, --help   help for completion

2020/11/04 07:50:14 accepts 1 arg(s), received 0
```

Could you please keep the same behaviour?

Reply:
Changed back to subcommands instead of flags for each shell so that iot provides better help message if the shell is missing